hrms/overrides/employee_advance.py

```diff
@@ -0,0 +1,269 @@
+                    None,
+                    TaskType.APPROVE_EMPLOYEE_ADVANCE,
+                    "Employee Advance",
+                    self.name,
+                    None,
+                    None,
+                    [self.approver_1],
+                )
+            elif self.status == EmployeeAdvanceConstants.PENDING_APPROVAL_BY_HR:
+                if old_doc.status not in [EmployeeAdvanceConstants.PENDING_APPROVAL]:
+                    frappe.throw(_(f"Invalid State Transition to state {self.state}"))
+                if not (
+                    frappe.session.user == self.approver_1
+                    or frappe.session.user == "Administrator"
+                ):
+                    frappe.throw(_("Only the Added approver can approve"))
+                CustomTask.mark_tasks_as_completed(
+                    "Employee Advance", self.name, TaskType.APPROVE_EMPLOYEE_ADVANCE
+                )
+            elif self.status == EmployeeAdvanceConstants.PENDING_PAYMENT:
+                if old_doc.status not in [EmployeeAdvanceConstants.PENDING_APPROVAL_BY_HR]:
+                    frappe.throw(_(f"Invalid State Transition to state {self.state}"))
+                if not user_has_role(frappe.session.user, RoleConstants.HR_L2_EXPENSE_ROLE):
+                    frappe.throw(_("Only user having HR L2 Expense Role can approve the document"))
+            elif self.status == EmployeeAdvanceConstants.PAID and self.docstatus == 1:
+                if old_doc.status not in [
+                    EmployeeAdvanceConstants.PENDING_PAYMENT,
+                    EmployeeAdvanceConstants.PARTLY_CLAIMED,
+                    EmployeeAdvanceConstants.CLAIMED,
+                ]:
+                    frappe.throw(_(f"Invalid State Transition to state {self.state}"))
+                if old_doc.status in [EmployeeAdvanceConstants.PENDING_PAYMENT]:
+                    if not user_has_role(frappe.session.user, RoleConstants.FINANCE_ROLE):
+                        frappe.throw(
+                            _(
+                                f"Invalid permission, only user with role {RoleConstants.FINANCE_ROLE} "
+                            )
+                        )
+            elif self.status == EmployeeAdvanceConstants.CANCELLED:
+                if old_doc.status not in [
+                    EmployeeAdvanceConstants.DRAFT,
+                    EmployeeAdvanceConstants.PENDING_PAYMENT,
+                    EmployeeAdvanceConstants.PENDING_APPROVAL_BY_HR,
+                ]:
+                    frappe.throw(_(f"Invalid State Transition to state {self.state}"))
+                if (
+                    old_doc.status == EmployeeAdvanceConstants.DRAFT
+                    and not frappe.session.user != self.owner
+                ):
+                    frappe.throw(_(f"Can be Canceled by Owner Only"))
+                elif (
+                    old_doc.status == EmployeeAdvanceConstants.PENDING_PAYMENT
```

Comment:
only admin can cancel from this status. For pending approval ur condition will work

Reply:
Done